src/vs/workbench/parts/terminal/electron-browser/terminalService.ts

```diff
@@ -222,14 +222,7 @@ export class TerminalService implements ITerminalService {
 	}
 
 	private createTerminalProcess(): ITerminalProcess {
-		let env = this.cloneEnv();
-		let shell = this.configHelper.getShell();
-		env['PTYPID'] = process.pid.toString();
-		env['PTYSHELL'] = shell.executable;
-		shell.args.forEach((arg, i) => {
-			env[`PTYSHELLARG${i}`] = arg;
-		});
-		env['PTYCWD'] = this.contextService.getWorkspace() ? this.contextService.getWorkspace().resource.fsPath : os.homedir();
+		let env = this.createTerminalEnv();
```

Comment:
May as well do the call inside `cp.fork` now. Other than that good to go though I'd like to see some tests for it since it see,s fairly testable.